File: app/lib/widgets/layout/device_list_panel_width.dart
/// Wide layout breakpoint — keep in sync with web `deviceListPanelWidth.ts`.
const double kDeviceListWideBreakpoint = 768;

/// Sidebar width at the minimum wide viewport (768px).
const double kDeviceListPanelMinWidth = 220;

/// Sidebar width cap on very large screens.
const double kDeviceListPanelMaxWidth = 264;

/// Fraction of viewport width used for the device list on wide layouts.
const double kDeviceListPanelWidthRatio = 0.205;

/// Absolute minimum when the user drags the divider.
const double kDeviceListPanelDragMinWidth = 220;

/// Absolute maximum when the user drags the divider.
const double kDeviceListPanelDragMaxWidth = 320;

/// Resolve device-list sidebar width from viewport width.
double resolveDeviceListPanelWidth(double viewportWidth) {
  if (viewportWidth < kDeviceListWideBreakpoint) {
    return viewportWidth;
  }
  final raw = viewportWidth * kDeviceListPanelWidthRatio;
  return raw.clamp(kDeviceListPanelMinWidth, kDeviceListPanelMaxWidth);
}

File: web/src/lib/deviceListPanelWidth.ts
/** Wide layout breakpoint — keep in sync with Tailwind `md` and Flutter `main_layout.dart`. */
export const DEVICE_LIST_WIDE_BREAKPOINT = 768;

/** Sidebar width at the minimum wide viewport (768px). */
export const DEVICE_LIST_PANEL_MIN_WIDTH = 220;

/** Sidebar width cap on very large screens. */
export const DEVICE_LIST_PANEL_MAX_WIDTH = 264;

/** Fraction of viewport width used for the device list on wide layouts. */
export const DEVICE_LIST_PANEL_WIDTH_RATIO = 0.205;

/**
 * Resolve device-list sidebar width from viewport width.
 * Narrow wide screens get a tighter list; larger screens get more room for names/status.
 */
export function resolveDeviceListPanelWidth(viewportWidth: number): number {
  if (viewportWidth < DEVICE_LIST_WIDE_BREAKPOINT) {
    return viewportWidth;
  }
  const raw = viewportWidth * DEVICE_LIST_PANEL_WIDTH_RATIO;
  return Math.round(
    Math.min(DEVICE_LIST_PANEL_MAX_WIDTH, Math.max(DEVICE_LIST_PANEL_MIN_WIDTH, raw)),
  );
}
